app/uk/gov/hmrc/emailverification/models/PasscodeRequest.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.emailverification.models
+
+import org.joda.time.format.ISOPeriodFormat
+import org.joda.time.Period
+import play.api.libs.json.{JsPath, Json, Reads}
+
+case class PasscodeRequest(sessionId:String,
```

Comment:
I'd argue that we should be getting the session ID from the request headers (since it should be there anyway)

I also think that it's not the right design decision to allow/require calling teams to provide the email template and link expiry duration - it would make more sense to me that these should be static, unless there's some specific requirement.

I.e. the request would just require the email.

I think we should discuss this in more detail with Andy

Reply:
yeah I guessed it was done like this so we don't own content that's part of someone elses journey and allows passcode email verification at different points with different messages so I was reluctant to change it but agree that's probably unnecessary